great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -759,16 +760,25 @@ def add_checkpoint(  # noqa: PLR0913
             checkpoint=checkpoint,
         )
 
+        result: Checkpoint | CheckpointConfig
```

Comment:
Should we maybe always return a `CheckpointConfig`? Feels a little odd that we have this union (but maybe a matter for another time)

Reply:
The return annotation says `-> Checkpoint` so I would expect to see a mypy error here.